client/core/core.go

```diff
@@ -7572,16 +7735,138 @@ func handleRedemptionRoute(c *Core, dc *dexConnection, msg *msgjson.Message) err
 	var oid order.OrderID
 	copy(oid[:], redemption.OrderID)
 
-	tracker, _, _ := dc.findOrder(oid)
-	if tracker == nil {
-		return fmt.Errorf("redemption request received for unknown order: %s", string(msg.Payload))
+	tracker, _, isCancel := dc.findOrder(oid)
+	if tracker != nil {
+		if isCancel {
+			return fmt.Errorf("redemption request received for cancel order %v, match %v (you ok server?)",
+				oid, redemption.MatchID)
+		}
+		err = tracker.processRedemption(msg.ID, redemption)
+		if err != nil {
+			return err
+		}
+		c.schedTradeTick(tracker)
+		return nil
 	}
-	err = tracker.processRedemption(msg.ID, redemption)
+
+	sigMsg := redemption.Serialize()
+	err = dc.acct.checkSig(sigMsg, redemption.Sig)
+	if err != nil {
+		c.log.Warnf("Server redemption signature error: %v", err) // just warn
+	}
```

Comment:
Does this need to be moved up now?

Reply:
Before the call to `processRedemption`? Yah, I suppose could do that, and remove the redundant `checkSig` from that method?

Reply:
Ah. I didn't notice the check in `processRedemption`. But yeah, better the way you have it in your latest commit.